src/main/scala/PropertyChecker.scala

```diff
@@ -402,4 +412,84 @@ object PropertyChecker {
        (objectNode, warnings, csvwPropertyType)
     }
   }
+
+  def tableSchemaProperty(typeString: PropertyType.Value): (JsonNode, String, String) => (JsonNode, Array[String], PropertyType.Value) = {
+    def tableSchemaPropertyInternal(value: JsonNode, baseUrl: String, lang: String): (JsonNode, Array[String], PropertyType.Value) = {
+      var schemaBaseUrl = new URL(baseUrl)
+      var schemaLang = lang
+
+      var schemaJson: ObjectNode = null
+      if(value.isTextual) {
+        val schemaUrl = new URL(new URL(baseUrl), value.asText())
+        val objectMapper = new ObjectMapper()
+        schemaJson = objectMapper.readTree(schemaUrl).asInstanceOf[ObjectNode]
+        if (!schemaJson.path("@id").isMissingNode) {
+          // Do something here as object node put method doesn't allow uri object
+          val absoluteSchemaUrl = new URL(schemaUrl, schemaJson.get("@id").asText())
+          schemaJson.put("@id", absoluteSchemaUrl.toString)
+        } else {
+          schemaJson.put("@id", schemaUrl.toString)
+        }
+        val (newSchemaBaseUrl, newSchemaLang) = fetchSchemaBaseUrlAndLangAndRemoveContext(schemaJson, schemaBaseUrl, schemaLang)
+        schemaBaseUrl = newSchemaBaseUrl
+        schemaLang = newSchemaLang
+      } else if(value.isObject) {
+        schemaJson = value.deepCopy()
+      } else {
+        return (NullNode.getInstance(), Array[String]("invalid_value"), PropertyType.Table)
+      }
+
+      var warnings = Array[String]()
+      val fieldsAndValues = Array.from(schemaJson.fields.asScala)
+      for(fieldAndValue <- fieldsAndValues) {
+        val p = fieldAndValue.getKey
+        val v = fieldAndValue.getValue
+        if(p == "@id") {
+          val matcher = PropertyChecker.startsWithUnderscore.pattern.matcher(v.asText())
+          if (matcher.matches) {
+            throw new MetadataError(s"@id ${v.asText} starts with _:")
+          }
+        } else if(p == "@type") {
+          if (v.asText() != "Schema") {
+            throw new MetadataError("@type of schema is not 'Schema'")
+          }
+        } else {
+          val (v1, warningsForP, typeString) = checkProperty(p, v, schemaBaseUrl.toString, schemaLang)
```

Comment:
Pick a better name than `v1` please. And yes, I know I did it.

Reply:
Also `typeString` again.

Reply:
changed names